catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImpl.java

```diff
@@ -66,147 +131,867 @@ public FederationAdminServiceImpl() {
+                    message,
+                    metacardId,
+                    sourceIds);
+            throw new FederationAdminException(message);
+        }
+
+        String xml = metacards.get(0)
+                .getMetadata();
+        return getRegistryPackageFromString(xml);
     }
 
     @Override
-    public List<Metacard> getRegistryMetacardsByRegistryIds(List<String> ids)
+    public void refreshRegistrySubscriptions() throws FederationAdminException {
+        if (CollectionUtils.isEmpty(pollableSourceIds)) {
+            return;
+        }
+
+        addOrUpdateMetacardsForRemoteRegistries();
+    }
+
+    public void init() {
+        try {
+            if (getRegistryIdentityMetacard() == null) {
+                createIdentityNode();
+            }
+        } catch (FederationAdminException e) {
+            LOGGER.error("There was an error bringing up the Federation Admin Service.", e);
+        }
+    }
+
+    public void bindRegistryStore(ServiceReference serviceReference) {
+        BundleContext bundleContext = getBundleContext();
+
+        if (serviceReference != null && bundleContext != null) {
+            RegistryStore registryStore =
+                    (RegistryStore) bundleContext.getService(serviceReference);
+
+            if (registryStore.isPullAllowed()) {
+                pollableSourceIds.add(registryStore.getId());
+                try {
+                    addOrUpdateMetacardsForRemoteRegistries(Collections.singleton(registryStore.getId()));
+                } catch (FederationAdminException e) {
+                    LOGGER.warn(
+                            "Trying to access the registry store before it has finished initializing. Couldn't update registry for new registry store (ID: {}). No actions required.",
+                            registryStore.getId());
+                }
+            }
+        }
+    }
+
+    public void unbindRegistryStore(ServiceReference serviceReference) {
+        BundleContext bundleContext = getBundleContext();
+
+        if (serviceReference != null && bundleContext != null) {
+            RegistryStore registryStore =
+                    (RegistryStore) bundleContext.getService(serviceReference);
+
+            pollableSourceIds.remove(registryStore.getId());
+        }
+    }
+
+    private Map<String, Metacard> getRemoteRegistryMetacardsMap(Set<String> sourceIds)
             throws FederationAdminException {
-        return new ArrayList<>();
+        Map<String, Metacard> remoteRegistryMetacards = new HashMap<>();
+
+        Filter filter =
+                FILTER_FACTORY.and(FILTER_FACTORY.like(FILTER_FACTORY.property(Metacard.CONTENT_TYPE),
+                        RegistryConstants.REGISTRY_NODE_METACARD_TYPE_NAME),
+                        FILTER_FACTORY.like(FILTER_FACTORY.property(Metacard.TAGS),
+                                RegistryConstants.REGISTRY_TAG));
+
+        QueryImpl query = new QueryImpl(filter);
+        query.setPageSize(PAGE_SIZE);
```

Comment:
Should we sort by last modified just in case some system has over a thousand registry items?  Hopefully anything over a 1000 would be very old and can be ignored.